firestore.rules

```diff
@@ -55,18 +63,21 @@ service cloud.firestore {
     }
 
     function anonymousCreate() {
-      return request.resource.data.run_key.size() > 10
+      return ('run_key' in request.resource.data)
+        && request.resource.data.run_key.size() > 10
         // Make sure anonymous answers don't have platform_ids
         && !('platform_id' in request.resource.data);
     }
 
     // Anyone can read any anonymous docs with a valid run_key
```

Comment:
This comment seems out of date.

Reply:
@kswenson Which comment are you referring to?

The one directly above `Anyone can read any anonymous docs with a valid run_key` is still valid. The only change I made was to make sure the resource.data (doc contents) has a run_key property before checking the size of the value of the property. The code actually worked before I think because firestore was automatically catching exceptions and treating them as false return values from functions.  But while testing with the emulator it seems to work differently.

Reply:
Unless I'm misreading it, the previous rule said that anyone with a runKey longer than 10 characters could read any document. With your change, anyone with a runKey longer than 10 characters _that matches the runKey of the document being requested_ can read it, which seems different than "anyone can read any anonymous document".

Reply:
The ` ('run_key' in request.resource.data)` means: "is there a key called 'run_key' in the object request.resource.data". So it isn't check check that the value of the run_key matches just that the key exists.